### ISSUE TITLE
Faulty header generation or parsing

### ISSUE THREAD

**Issue by sandfox** [flowtype/flow-typed#923]
Short version:

One of the flow-type defs installed/generated during `flow-typed install` has the following header
```
// flow-typed signature: 53b581d81115b53eaf3a74ae963c4f43
// flow-typed version: ba3c3d2dca/lodash_v4.x.x/flow_>=v0.38.x <=v0.46.x
```

Subsequently, when running `flow-typed install` again the cli errors out

```
UNCAUGHT ERROR: Error: flow_v0.38.x <=v0.46.x: Unexpected trailing characters:  <=v0.46.x
    at validationError (/app/node_modules/flow-typed/dist/lib/validationErrors.js:14:11)
    at parseDirString (/app/node_modules/flow-typed/dist/lib/flowVersion.js:155:45)
    at _callee2$ (/app/node_modules/flow-typed/dist/lib/npm/npmLibDefs.js:502:146)
    at tryCatch (/app/node_modules/regenerator-runtime/runtime.js:64:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/app/node_modules/regenerator-runtime/runtime.js:355:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/app/node_modules/regenerator-runtime/runtime.js:116:21)
    at tryCatch (/app/node_modules/regenerator-runtime/runtime.js:64:40)
    at invoke (/app/node_modules/regenerator-runtime/runtime.js:154:20)
    at /app/node_modules/regenerator-runtime/runtime.js:164:13
```

We are running flow-typed  "2.1.2"
I've done some basic spelunking around with a colleague and our very vague guesswork is that the cause maybe lies in `cli/src/lib/flowVersion.js`

It looks the version string is created by `toSemverString` and explicitly includes a space, but then the header is parsed with `parseDirString`, which does not expect spaces, and instead expects the versions to be separated by a `-`.  Maybe the parser is expecting the header version string to be written using `toDirString` instead, or maybe there should be a `parseSemverString` method?

**Comment by eric-burel** [comment]
Same issue here, is this related to a recent update ?

**Comment by sandfox** [comment]
@eric-burel very much in dangerously little knowledge territory here, but I think this particular bug only manifests itself for certain sets of dependencies, i.e the installed dependency graph of your app at a particular time. There may also be other factors that influence this such as flow-typed version and lunar cycle etc. (caveats, maybe we're seeing different causes of the same symptom?)

**Comment by schovi** [comment]
Same error for me. Cleaning up node_modules, or ~/.flow-typed dont help at all.

**Comment by BowlingX** [comment]
same here, I get `UNCAUGHT ERROR: Error: flow_v0.23.x_<=v0.27.x: Unexpected trailing characters: _<=v0.27.x`.

**Comment by flessard** [comment]
Same error for me. Error: flow_v0.38.x <=v0.46.x: Unexpected trailing characters:  <=v0.46.x

**Comment by eric-burel** [comment]
Not that I couldn't found those faulty headers in any files, even in the `~/.flow-typed` folder, I can't tell where it comes from. Is it dynamically generated ?

**Comment by admire93** [comment]
Installing flow-typed again after delete all `flow` `flow-typed` `~/.flow-typed` directory works for me.

**Comment by augbog** [comment]
Deleting the `.flow-typed` directory didn't work for me :/

**Comment by DrummerHead** [comment]
I got:

```
UNCAUGHT ERROR: Error: flow_v0.30.x <=v0.52.x: Unexpected trailing characters:  <=v0.52.x
```

PRELUDE: I have a `postinstall` script on `package.json` that executes `flow-typed install` (installed locally) that would look like:

```
  "scripts": {
    "postinstall": "flow-typed install",
  }
```

Then I ran:

```
rm -rf node_modules/flow-typed/
rm -rf flow-typed/
yarn add flow-typed
yarn postinstall
```

which made:
- flow-typed update from "^2.1.2" to "^2.1.5"
- yarn.lock changes
- several `flow-typed` definitions to update

This fixed the issue :) (but now flow doesn't recognize jest global functions... but that's a new error...)

**Comment by jedwards1211** [comment]
Guys, you may want to be careful with just blowing away the `flow-typed` directory.  It's designed for you to be able to edit it (in case you want to add missing defs for anything) and as such someone else in your org may have added their own defs in there.

**Comment by dreid** [comment]
This should be fixed in v2.2 by #1311
